controllers/limit/hardwareLimitUpdater.go

```diff
@@ -0,0 +1,141 @@
+package limit
+
+import (
+	"context"
+	"fmt"
+	"github.com/cloudogu/cesapp-lib/core"
+	"github.com/cloudogu/cesapp-lib/registry"
+	k8sv1 "github.com/cloudogu/k8s-dogu-operator/api/v1"
+	coreosclient "github.com/coreos/etcd/client"
+	"github.com/hashicorp/go-multierror"
+	v1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	triggerSyncEtcdKeyFullPath = "/config/_global/trigger-container-limit-sync"
+)
+
+// hardwareLimitUpdater is responsible to update all hardware limits for dogu deployments when a certain trigger is called.
+type hardwareLimitUpdater struct {
+	client           client.Client
+	namespace        string
+	registry         registry.Registry
+	doguLimitPatcher limitPatcher
+}
+
+type limitPatcher interface {
+	// RetrieveMemoryLimits reads all container keys from the dogu configuration and creates a DoguLimits object.
+	RetrieveMemoryLimits(doguResource *k8sv1.Dogu) (DoguLimits, error)
+	// PatchDeployment patches the given deployment with the resource limits provided.
+	PatchDeployment(deployment *v1.Deployment, limits DoguLimits) error
+}
+
+// DoguLimits contains all data necessary to limit the physical resources for a dogu.
+type DoguLimits struct {
+	// Sets the cpu requests and limit values for the dogu deployment to the contained value. For more information about resource management in Kubernetes see https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/.
+	CpuLimit string
+	// Sets the memory requests and limit values for the dogu deployment to the contained value. For more information about resource management in Kubernetes see https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/.
+	MemoryLimit string
+	// Sets the ephemeral storage requests and limit values for the dogu deployment to the contained value. For more information about resource management in Kubernetes see https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/.
+	EphemeralStorageLimit string
+}
+
+// NewHardwareLimitUpdater creates a new runnable responsible to detect changes in the container configuration of dogus.
+func NewHardwareLimitUpdater(client client.Client, namespace string) (*hardwareLimitUpdater, error) {
+	endpoint := fmt.Sprintf("http://etcd.%s.svc.cluster.local:4001", namespace)
```

Comment:
we could use the registry from main here

Reply:
We could but there are no real benefits, right?